packages/pipeline/src/utils/get_ohlcv_trading_pairs.ts

```diff
@@ -0,0 +1,100 @@
+import axios from 'axios';
+import { chain } from 'ramda';
+import { Connection } from 'typeorm';
+
+export interface TradingPair {
+    fromSymbol: string;
+    toSymbol: string;
+    latest: number;
+}
+
+interface StaticPair {
+    fromSymbol: string;
+    toSymbol: string;
+}
+
+const COINLIST_API = 'https://min-api.cryptocompare.com/data/all/coinlist?BuiltOn=7605';
+
+interface CryptoCompareCoinListResp {
+    Data: Map<string, CryptoCompareCoin>;
+}
+
+interface CryptoCompareCoin {
+    Symbol: string;
+    BuiltOn: string;
+    SmartContractAddress: string;
+}
+
+const TO_CURRENCIES = ['USD', 'EUR', 'ETH'];
+
+/**
+ * Get trading pairs with latest scraped time for OHLCV records
+ * @param conn a typeorm Connection to postgres
+ */
+export async function fetchOHLCVTradingPairs(
+    conn: Connection,
+    source: string,
+    earliestBackfillTime: number,
+): Promise<TradingPair[]> {
+    const rawTokenAddresses: Array<{ tokenaddress: string }> = await conn.query(
+        `SELECT DISTINCT(maker_token_address) as tokenaddress FROM raw.exchange_fill_events UNION
+        SELECT DISTINCT(taker_token_address) as tokenaddress FROM raw.exchange_fill_events
+        LIMIT 1`,
+    );
+    const tokenAddresses = rawTokenAddresses.map(obj => obj.tokenaddress);
+
+    // get token symbols used by Crypto Compare
+    const allCoins = await axios.get<CryptoCompareCoinListResp>(COINLIST_API);
+    const erc20Coins: Map<string, string> = new Map();
+    Object.entries(allCoins.data.Data).forEach(pair => {
+        const [k, v] = pair;
+        if (v.BuiltOn === '7605' && v.SmartContractAddress !== 'N/A') {
+            erc20Coins.set(v.SmartContractAddress, k);
+        }
+    });
+
+    const mapFn = (tokenAddress: string): StaticPair[] => {
+        return chain(fiat => {
+            const fromSymbol = erc20Coins.get(tokenAddress);
+            if (!!fromSymbol) {
+                return [
+                    {
+                        toSymbol: fiat,
+                        fromSymbol,
+                    },
+                ];
+            } else {
+                return [];
+            }
+        }, TO_CURRENCIES);
+    };
+    const tradingPairs: StaticPair[] = chain(mapFn, tokenAddresses);
+
+    const tradingPairsLatestQuery: string = tradingPairs
+        .map(pair => {
+            return `SELECT
+              case COUNT(*) when 0 then ${earliestBackfillTime} else MAX(end_time) end AS latest,
```

Comment:
How does CC deal with start time /etc? Will this leave hour gaps or cause hour overlaps?

Reply:
I refactored to a different implementation, but still a good question. I think the new logic is good. The script scrapes CC and errs on the side of overlapping data, and then we filter on save. 

Results:
![screen shot 2018-12-04 at 11 17 28 am](https://user-images.githubusercontent.com/8582774/49467011-4ba4cc00-f7b6-11e8-809e-d042b3e2d824.png)

810 records means 809 divisors. 
2912400000 ms/ 809 = 3,600,000 ms = 3600 s = 1hr

So I'm reasoning that we aren't getting any gaps or overlaps.